test/uk/gov/hmrc/tai/connectors/IabdConnectorSpec.scala

```diff
@@ -34,51 +34,128 @@ import scala.util.Random
+          val result = Await.result(sut.iabds(nino, TaxYear().next.next), 5.seconds)
+
+          result mustBe Json.arr()
+        }
       }
+    }
+
+    "toggled to use DES" must {
 
-      "looking for cy+2 year" in {
+      "return IABD json" in {
         val mockHttpHandler = mock[HttpHandler]
-        val mockConfig = mock[NpsConfig]
+        val mockNpsConfig = mock[NpsConfig]
+        val mockDesConfig = mock[DesConfig]
         val iabdUrls = mock[IabdUrls]
-        when(iabdUrls.iabdUrl(any(), any())).thenReturn("URL")
-        when(mockConfig.originatorId).thenReturn("TEST")
-        when(mockHttpHandler.getFromApi(Matchers.eq("URL"), Matchers.eq(APITypes.NpsIabdAllAPI))(Matchers.
+        val featureTogglesConfig = mock[FeatureTogglesConfig]
+
+        when(featureTogglesConfig.desEnabled).thenReturn(true)
+        when(iabdUrls.desIabdUrl(any(), any())).thenReturn("URL")
+        when(mockDesConfig.originatorId).thenReturn("TEST")
+        when(mockHttpHandler.getFromApi(Matchers.eq("URL"), Matchers.eq(APITypes.DesIabdAllAPI))(Matchers.
           eq(hc.withExtraHeaders("Gov-Uk-Originator-Id" -> "TEST")))).thenReturn(Future.successful(json))
-        val sut = createSUT(config = mockConfig, iabdUrls = iabdUrls, httpHandler = mockHttpHandler)
 
-        val result = Await.result(sut.iabds(nino, TaxYear().next.next), 5.seconds)
+        val sut = createSUT(mockNpsConfig, mockDesConfig, mockHttpHandler, iabdUrls, featureTogglesConfig)
+        val result = Await.result(sut.iabds(nino, TaxYear()), 5.seconds)
+
+        result mustBe json
+      }
 
-        result mustBe Json.arr()
+      "return empty json" when {
+        "looking for next tax year" in {
+          val mockHttpHandler = mock[HttpHandler]
+          val mockNpsConfig = mock[NpsConfig]
+          val mockDesConfig = mock[DesConfig]
+          val iabdUrls = mock[IabdUrls]
+          val featureTogglesConfig = mock[FeatureTogglesConfig]
+
+          when(iabdUrls.desIabdUrl(any(), any())).thenReturn("URL")
+          when(featureTogglesConfig.desEnabled).thenReturn(true)
+          when(mockDesConfig.originatorId).thenReturn("TEST")
+          when(mockHttpHandler.getFromApi(Matchers.eq("URL"), Matchers.eq(APITypes.DesIabdAllAPI))(Matchers.
+            eq(hc.withExtraHeaders("Gov-Uk-Originator-Id" -> "TEST")))).thenReturn(Future.successful(json))
+
+          val sut = createSUT(mockNpsConfig, mockDesConfig, mockHttpHandler, iabdUrls, featureTogglesConfig)
+          val result = Await.result(sut.iabds(nino, TaxYear().next), 5.seconds)
+
+          result mustBe Json.arr()
+        }
+
+        "looking for cy+2 year" in {
+          val mockHttpHandler = mock[HttpHandler]
+          val mockNpsConfig = mock[NpsConfig]
+          val mockDesConfig = mock[DesConfig]
+          val iabdUrls = mock[IabdUrls]
+          val featureTogglesConfig = mock[FeatureTogglesConfig]
+
+          when(iabdUrls.desIabdUrl(any(), any())).thenReturn("URL")
```

Comment:
It looks like the response from this mock is the same in every test, is it relevant - could it be moved to the bottom of the file and used for all tests?